providers/amazon/src/airflow/providers/amazon/aws/transfers/redshift_to_s3.py

```diff
@@ -102,7 +103,7 @@ def __init__(
         table: str | None = None,
         select_query: str | None = None,
         redshift_conn_id: str = "redshift_default",
-        aws_conn_id: str | None = "aws_default",
+        aws_conn_id: str | None | ArgNotSet = NOTSET,
```

Comment:
Why not just set the default value to `None`? What will `S3Hook.get_connection(conn_id=None)` return?

Reply:
Because I'm trying to keep a couple things true at the same time:

1) Don't make a breaking change, this is basically the tl;dr answer to your question. So far this operator has been inserting that connection id, and there could be many dags out there that are happily working with this default. So if I now make it None, that will break for those users.
2) None is a value the user could supply, and I'm really trying to detect the case where they don't supply anything. (but again 1/ is mostly the issue).

Reply:
I understand your arguments, but when the user provides a `None` value, `conn_not_set` will be `False` and `_aws_conn_id` will be `None`, then `conn = S3Hook.get_connection(conn_id=self._aws_conn_id) if not self.conn_not_set else None` will be evaluated as `conn = S3Hook.get_connection(conn_id=None)` which is not the case today, and I don't know if in this case, the operator will fail.

Reply:
Just checked, and found that because of:
```
        try:
            uri = SecretCache.get_connection_uri(conn_id)
            return Connection(conn_id=conn_id, uri=uri)
        except SecretCache.NotPresentException:
            pass  # continue business
```
conn will be `None` and the code in else block will be executed.